infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDetector.java

```diff
@@ -153,6 +149,9 @@ private PollReport pollRound(long epoch,
         // In each iteration we poll all the servers in the members list.
         for (int iteration = 0; iteration < failureThreshold; iteration++) {
 
+            final long iterationStartTime = System.nanoTime();
+            long interIterationSleepTime = interIterationInterval;
+
```

Comment:
make interIterationInterval final and make sure its >1ms, otherwise you can have the opposite behavior of what this patch is trying to do. In linux, I don't think you can put a thread to sleep for less than 1ms, so if it gets downcast to 0, the thread won't sleep. If you need sub-millisecond precision you can use busy-waiting, but the sleep utils used here to even use that.

Reply:
interIterationInterval has setter/getters and currently being used. Its default is set at 1ms and the patch keeping max of the default and potentially increased time. so it won't go bellow 1ms except if it is specifically set through the setter.